packages/bun-usockets/src/internal/internal.h

```diff
@@ -122,6 +126,20 @@ struct us_socket_t {
                          = was in low-prio queue in this iteration */
   struct us_socket_context_t *context;
   struct us_socket_t *prev, *next;
+  struct us_connecting_socket_t *connect_state;
+};
+
+struct us_connecting_socket_t {
+    alignas(LIBUS_EXT_ALIGNMENT) void *addrinfo_req;
+    struct us_socket_context_t *context;
+    struct us_connecting_socket_t *next;
+    struct us_socket_t *socket;
+    int options;
+    int socket_ext_size;
+    unsigned int closed : 1, shutdown : 1, ssl : 1, shutdown_read : 1;
+    unsigned char timeout;
+    unsigned char long_timeout;
+    int error;
```

Comment:
does the error need to be stored?

Reply:
it's necessary so `us_connecting_socket_get_error` can return the right error if `getaddrinfo` fails, as `getaddrinfo` doesn't set `errno`